doit/cmd_resetdep.py

```diff
@@ -0,0 +1,74 @@
+from .cmd_base import DoitCmdBase, check_tasks_exist
+from .cmd_base import subtasks_iter
+
+
+class ResetDep(DoitCmdBase):
+    name = "reset-dep"
+    doc_purpose = ("recompute and save the state of file dependencies without "
+                   "executing actions")
+    doc_usage = "[TASK ...]"
+    cmd_options = ()
+    doc_description = """
+This command allows to recompute the informations on file dependencies
+(timestamp, md5sum, ... depending on the ``check_file_uptodate`` setting), and
+save this in the database, without executing the actions.
+
+The command run on all tasks by default, but it is possible to specify a list
+of tasks to work on.
+
+This is useful when the targets of your tasks already exist, and you want doit
+to consider your tasks as up-to-date. One use-case for this command is when you
+change the ``check_file_uptodate`` setting, which cause doit to consider all
+your tasks as not up-to-date. It is also useful if you start using doit while
+some of your data as already been computed, or when you add a file dependency
+to a task that has already run.
+"""
+
+    def _execute(self, pos_args=None):
+        filter_tasks = pos_args
+
+        # dict of all tasks
+        tasks = dict([(t.name, t) for t in self.task_list])
+
+        if filter_tasks:
+            # list only tasks passed on command line
+            check_tasks_exist(tasks, filter_tasks)
+            # get task by name
+            task_list = []
+            for name in filter_tasks:
+                task = tasks[name]
+                task_list.append(task)
+                task_list.extend(subtasks_iter(tasks, task))
+        else:
+            task_list = self.task_list
+
+        write = self.outstream.write
+        for task in task_list:
+            # Save this now because dep_manager.get_status will remove the task
+            # from the db if the checker changed.
+            task.values = self.dep_manager.get_values(task.name)
+            task.results = self.dep_manager.get_results(task.name)
+
+            try:
+                run_status = self.dep_manager.get_status(task, tasks)
+            except Exception:
```

Comment:
Actaully I dont get this. If it fails here it will also fail when you call `save_success()` so you could write the error message and `continue` from here.